twoliter/embedded/Makefile.toml

```diff
@@ -354,11 +354,29 @@ dependencies = ["setup"]
 script_runner = "bash"
 script = [
 '''
+declare -a CARGO_MANIFESTS
+declare -a CARGO_DIGESTS
+mkdir -p .cargo/vendor
+if [ ! -f ".cargo/config.toml" ]; then
+    touch .cargo/config.toml
+fi
 for ws in sources variants .; do
   [ -s "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml" ] || continue
-  cargo fetch --locked --manifest-path "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml"
+  CARGO_MANIFESTS+=(--sync "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml")
+  CARGO_DIGESTS+=("$(sha256sum "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.lock" | awk '{ print $1 }')")
 done
-
+DIGEST="$(echo ${CARGO_DIGESTS[@]} | sha256sum | awk '{ print $1 }')"
+MARKER_FILE=".cargo/vendor-${DIGEST}.marker"
+if [ -f "${MARKER_FILE}" ]; then
+    echo "no changes detected in cargo projects, skipping cargo vendor"
+else
+    cargo vendor --versioned-dirs --respect-source-config --locked ${CARGO_MANIFESTS[@]} .cargo/vendor > .cargo/config.toml 2> .cargo/vendor.log || {
```

Comment:
If the marker and log file logic aren't essential, I'd prefer to omit them and stay closer to the previous implementation:
```
- cargo fetch --locked --manifest-path "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml"
+ cargo vendor --locked --sync "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml" --no-delete  --versioned-dirs --respect-source-config  --quiet .cargo/vendor
```

Reply:
The markerfile was done to prevent running cargo vendor every time you call twoliter build commands. It was meant to optimize timing as when you cargo vendor it does the whole shebam over.

Reply:
Removed the marker file